packages/amplify-codegen/src/commands/models.js

```diff
@@ -91,9 +90,7 @@ async function generateModels(context, generateOptions = null) {
   });
 
   const schemaContent = loadSchema(apiResourcePath);
-
-  const baseOutputDir = path.join(projectRoot, getModelOutputPath(context));
-  const schema = parse(schemaContent);
+  const baseOutputDir = overrideOutputDir || path.join(projectRoot, getModelOutputPath(context));
```

Comment:
weren't we respecting the `overrideOutputDir` before?

Reply:
The graphql-generator is returning the generated assets with just the base filename. Then the directory is joined with the filename when writing (line 130).

This is an option I'll be bringing up in the API review. Either:
1. pass a directory to graphql-generator as a param
2. Whatever calls graphql-generator (amplify-codegen/CLI) will be responsible for joining the correct directories when writing.

Reply:
overrideOutputDir should be a concern of the amplify CLI, not of whatever this vNew utility will be, nor the generator. `generate vNext` will have it's own mechanism for communicating output paths, but again, that's core concern of that separate layer.